lib/3scale/client.rb

```diff
@@ -52,24 +52,37 @@ class Client
     private_constant :EXTENSIONS_HEADER
 
     def initialize(options)
-      if options[:provider_key].nil? || options[:provider_key] =~ /^\s*$/
-        raise ArgumentError, 'missing :provider_key'
+      @provider_key = options[:provider_key]
+      @service_token = options[:service_token]
+      @use_service_tokens = options[:use_service_tokens]
```

Comment:
Why not just accept provider_key or service_token and use whatever was passed ? And raise if none was.